packages/material-ui-lab/src/SpeedDialAction/SpeedDialAction.js

```diff
@@ -37,13 +37,26 @@ class SpeedDialAction extends React.Component {
   };
 
   handleTooltipClose = () => {
+    if (this.props.tooltipOpen) return;
     this.setState({ tooltipOpen: false });
   };
 
   handleTooltipOpen = () => {
+    if (this.props.tooltipOpen) return;
     this.setState({ tooltipOpen: true });
   };
 
+  componentDidUpdate = prevProps => {
+    if (!this.props.tooltipOpen || prevProps.open === this.props.open) return;
+    if (!this.props.open && this.state.tooltipOpen) {
+      this.setState({ tooltipOpen: false });
```

Comment:
Do you see a way to move this to `getDerivedStateFromProps` and save a rendercycle?

Reply:
done